src/components/SearchBar.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from "react";
+
+// TODO: Fix query reset
+
+interface ISearchBarState {
+    query: string;
+}
+
+class SearchBar extends React.Component<{}, ISearchBarState> {
+    state = {
+        query: null,
+    };
+
+    checkForEnter(event) {
+        const enterKeyCode = 13;
+        if ( event.keyCode !== enterKeyCode ) {
+            return;
+        }
+
+        this.querySubmit();
+    }
+
+    queryUpdate(event) {
+        this.setState( { query: event.target.value } );
+    }
+
+    querySubmit() {
+        if ( this.state.query == null ) {
+            return;
+        }
+
+        alert("Submitting query: ".concat(this.state.query));
+
+        this.setState( {query: null } );
+    }
+
+    render() {
+        return (
+        <div className="input-group">
+            <input type="text" className="form-control" placeholder="Search..."
+                   value={this.state.query}
+                   onChange={this.queryUpdate.bind(this)}
```

Comment:
I believe if you make the `queryUpdate` and `checkForEnter` as arrow functions you won't need to bind anymore